File: sports-news-web/src/api/address.ts
import request from '../utils/request'

// 获取地址列表
export const getAddressList = () => {
  return request({
    url: '/address/list',
    method: 'get'
  })
}

// 添加地址
export const addAddress = (data: any) => {
  return request({
    url: '/address',
    method: 'post',
    data
  })
}

// 更新地址
export const updateAddress = (data: any) => {
  return request({
    url: '/address',
    method: 'put',
    data
  })
}

// 删除地址
export const deleteAddress = (id: number) => {
  return request({
    url: `/address/${id}`,
    method: 'delete'
  })
}

// 地址接口参数类型定义
export interface AddressForm {
  id?: number
  receiver: string
  phone: string
  province: string
  city: string
  district: string
  detail: string
  isDefault: number
} 

File: src/main/java/com/news/sports/entity/Address.java
package com.news.sports.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("address")
public class Address {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long userId;
    
    private String receiver;
    
    private String phone;
    
    private String province;
    
    private String city;
    
    private String district;
    
    private String detail;
    
    private Integer isDefault;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
    
    @TableLogic
    private Integer deleted;
} 